main.go

```diff
@@ -114,7 +114,7 @@ func main() {
 	case "aws":
 		p, err = provider.NewAWSProvider(cfg.DomainFilter, cfg.DryRun)
 	case "inmemory":
-		p, err = provider.NewInMemoryProviderWithDomainAndLogging("example.com"), nil
+		p, err = provider.NewInMemoryProvider(provider.InMemoryWithDomain(cfg.DomainFilter), provider.InMemoryWithLogging()), nil
```

Comment:
The benefit is now to have tons of initialisers, which are purposed to override default behaviour. Some fields are not exposed and exposing them would not be beneficial for the users. However this fields may be used to provide certain features. In this case providing user with a single interface to override those default by optional "options" is a possible solution.